google/cloud/firestore_v1/rate_limiter.py

```diff
@@ -17,7 +17,7 @@
 
 
 def utcnow():
-    return datetime.datetime.utcnow()
+    return datetime.datetime.now(tz=datetime.timezone.utc)
```

Comment:
```suggestion
    # We used datetime.utcnow() before, since it's deprecated from python 3.12,
    # we are using datetime.now(timezone.utc) now. "utcnow()" is offset-native
    # (no timezone info), but "now()" is offset-aware (with timezone info).
    # This will cause datetime comparison problem. For backward compatibility,
    # we need to remove the timezone info.
    now = datetime.datetime.now(datetime.timezone.utc)
    now = now.replace(tzinfo=None)
    return now
```

Recommending that we make the same change as `google-auth` for backward compatibility
https://github.com/googleapis/google-auth-library-python/pull/1413/files

Reply:
My initial thought is that since the method name doesn't have a single leading underscore it could have been used. I think we should ensure backward compatibility. For example, it's `def utcnow():` and not `def _utcnow():`

https://peps.python.org/pep-0008/#method-names-and-instance-variables
`Use one leading underscore only for non-public methods`

Reply:
I was just going off of Mariatta's comment here, since I raised the same concern initially: https://github.com/googleapis/python-firestore/pull/716#discussion_r1191365451

It is arguable that this function is public, though also it suffers from the same problems that `datetime.utcnow` suffers from, so if you're going to have it continue to return a naïve datetime, it should probably have a deprecation warning anyway.

The linked commit definitely seems to be doing the wrong thing, since it's re-creating `datetime.utcnow` in a private function.

Reply:
I don't have the full context on the issue, but this `utcnow` function is just a one-liner, and only used in a few places. Can we mark it as deprecated and just replace its internal usages with `datetime.datetime.now(datetime.timezone.utc)?`

I feel like keeping it around is more trouble than its worth

Reply:
Deprecated `google.cloud.firestore_v1.rate_limiter.utcnow` in 075e999a4638fd137e65757f8d5a65a3aa3fcef9